File: ruby/easy/detect_capital.rb
# @param {String} word
# @return {Boolean}
def detect_capital_use(word)
    for i in 0..word.length-2
        return false if (!is_caps(word[i]) && is_caps(word[i+1])) || (is_caps(word[i]) && !is_caps(word[i+1]) && i > 0)
    end
    true
end

def is_caps(c)
    return c >= 'A' && c <= 'Z';
end

File: javascript/easy/detectCapital.js
/**
 * @param {string} word
 * @return {boolean}
 */
var detectCapitalUse = function(word) {
    for(let i = 0; i < word.length-1; i++) {
        if((!isCaps(word[i]) && isCaps(word[i+1])) || (isCaps(word[i]) && !isCaps(word[i+1]) && i > 0)) {
            return false;
        }
    }
    return true;
};

var isCaps = function(c) {
    return c.charCodeAt(0) >= 65 && c.charCodeAt(0) < 97;    
};
